src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder_ListPatterns.cs

```diff
@@ -10,40 +10,38 @@ namespace Microsoft.CodeAnalysis.CSharp
 {
     internal sealed partial class DecisionDagBuilder
     {
-        private void MakeTestsAndBindingsForLengthAndListPatterns(BoundDagTemp input, BoundRecursivePattern recursive, ArrayBuilder<BoundPatternBinding> bindings, ArrayBuilder<Tests> tests)
+        private Tests MakeTestsAndBindingsForListPattern(BoundDagTemp input, BoundListPattern list, out BoundDagTemp output, ArrayBuilder<BoundPatternBinding> bindings)
         {
-            if (recursive.HasErrors)
-                return;
+            Debug.Assert(input.Type.IsErrorType() || list.HasErrors || list.InputType.IsErrorType() ||
```

Comment:
I would have thought `input.Type.Equals(list.InputType, TypeCompareKind.AllIgnoreOptions)` was always true. When is it not the case?
Could we tighten from `AllIgnoreOptions` to `ConsiderEverything`?

Reply:
>I would have thought input.Type.Equals(list.InputType, TypeCompareKind.AllIgnoreOptions) was always true.

I think so. This is the assert we have for recursive patterns.